lib/connectors/memory.js

```diff
@@ -421,6 +421,11 @@ function applyFilter(filter) {
       // Also support array types. Mongo, possibly PostgreSQL
       if (Array.isArray(value)) {
         var matcher  = where[key];
+        // The following condition is for the case where we are querying with
+        // a neq filter, and when the value is an empty array ([]).
+        if (matcher.neq !== undefined && value.length <= 0) {
```

Comment:
In the scope of my correction, yes. My correction only deals with the empty array ([]). If you query with a 'neq' filter, the empty array should always match. I supposed that the preceding code works for cases that are not with the empty array. I can check that ...